types/react-router/index.d.ts

```diff
@@ -64,7 +64,7 @@ export interface RouteComponentProps<P> {
 
 export interface RouteProps {
   location?: H.Location;
-  component?: React.SFC<RouteComponentProps<any> | undefined> | React.ComponentClass<RouteComponentProps<any> | undefined>;
+  component?: React.SFC<RouteComponentProps<any> & any | undefined> | React.ComponentClass<RouteComponentProps<any> & any | undefined>;
```

Comment:
Adding ```& any ``` to a type erases it entirely so you might as well write this as:

```ts
component?: React.SFC<any> | React.ComponentClass<any>; 
```

Is this what you meant to do?